### PR TITLE
Csm 124 - Added the resource for the Chart Search Note 

### PR DESCRIPTION
**Pull request by ssmusoke**

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * REST resource representing a {@link ChartSearchNote}.
+ */
+@Resource(name = RestConstants.VERSION_1 + "/chartSearchNote", supportedClass = ChartSearchBookmark.class, supportedOpenmrsVersions = {"1.10.x", "1.11.*", "1.12.*", "2.0.*"})
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java]: The convention is to make the resource name all lowercase

File: omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+public class ChartSearchNoteResource extends DelegatingCrudResource<ChartSearchNote> {
+    @Override
+    public ChartSearchNote getByUniqueId(String uuid) {
+        return Context.getService(ChartSearchService.class).getSearchNoteByUuid(uuid);
+    }
+
+    @Override
+    protected void delete(ChartSearchNote chartSearchNote, String s, RequestContext requestContext) throws ResponseException {
+
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java]: I would assume this should void the note

File: omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Override
+    public void purge(ChartSearchNote chartSearchNote, RequestContext requestContext) throws ResponseException {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java]: Shouldn't this delete the note from the DB or it is not supported?

File: omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java (hunk 2 above)
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java]: Apparently this is not voidable, ignore my comment please

File: omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+
+        if (representation instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("comment");
+            description.addProperty("searchPhrase");
+            description.addProperty("patient", Representation.DEFAULT);
+            description.addProperty("noteOwner", Representation.DEFAULT);
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/resources/ChartSearchNoteResource.java]: Might be better to make get REF representations of patient and noteOwner

**Comment by dkayiwa** [comment]
@ssmusoke do you think you still have time to squash these commits into one? :)

**Comment by ssmusoke** [comment]
@dkayiwa - let me quash them before the end of day today

Stephen S. Musoke
Skype: ssmusoke
Twitter: @ssmusoke
+256 776 553 260
http://ssmusoke.com
*Our deepest fear is not that we are inadequate. Our deepest fear is that
we are powerful beyond measure ~Marianne Williamson*

**Pull request closed without merging**